executor_manager/routers/routers.py

```
@@ -1576,9 +1595,28 @@ async def restore_executor_workspace(
 
     except HTTPException:
         raise
+    except httpx.TimeoutException:
+        logger.error(f"[Restore] Upstream timeout for task {request.task_id}")
+        raise HTTPException(
+            status_code=504, detail="Executor restore service timed out"
+        )
```

Ruff (0.15.15)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)